tasking/pulp/tasking/base.py

```diff
@@ -388,7 +388,7 @@ def cancel(task_id):
         # If the task is already done, just stop
         msg = _('Task [%(task_id)s] already in a completed state: %(state)s')
         _logger.info(msg % {'task_id': task_id, 'state': task_status.state})
-        return
+        raise PulpException(msg % {'task_id': task_id, 'state': task_status.state})
```

Comment:
I do like telling the use exactly what's going on. But I do see your point about maybe just not worrying about this and telling the user we're making a best effort instead